### PR TITLE
[COM-120] Rendering Multithreading Initial Setup

### PR DESCRIPTION
**Pull request by Tissure**

### CODE REVIEW

**Comment by Tissure** [comment]
I give up on the scene graph copy for now. 
Race condition with scene graph exists, not currently noticeable.

File: Core/src/main.cpp [hunk 1]
```diff
@@ -18,57 +18,49 @@
 #include "System.h"
 #include "UI.h"
 #include "Utils/IDUtils.h"
+#include "Window/Window.h"
 
-void framebuffer_size_callback(GLFWwindow* window, int width, int height)
-{
-    glViewport(0, 0, width, height);
-}
+//void framebuffer_size_callback(GLFWwindow* window, int width, int height)
```
Comment (le-rich) [inline Core/src/main.cpp:23]: Don't leave dead code

File: Core/src/main.cpp [hunk 2]
```diff
@@ -91,63 +83,54 @@ int run_glfw()
...
 
-        previousTime = currentTime;
+		previousTime = currentTime;
 
         fixedUpdateBuffer += std::chrono::duration_cast<std::chrono::milliseconds>(deltaTime).count();
         std::cout << "COUNT: " << fixedUpdateBuffer << "\n"; // Commenting this line causes the Fixed update loop to stutter.
 
         Input::RunInputListener();
-        glfwMakeContextCurrent(window);
+        //glfwMakeContextCurrent(window);
```
Comment (le-rich) [inline Core/src/main.cpp:120]: Dead code

File: Core/src/main.cpp [hunk 3]
```diff
@@ -91,63 +83,54 @@ int run_glfw()
...
+		window.SetWindowToCurrentThread();
         // Rendering related calls, we can move these to the loop of the rendering thread
-        glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
-        glEnable(GL_DEPTH_TEST);
-        renderer->Update(deltaTimeFloatSeconds); // draw tri or square
+        //glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+        //glEnable(GL_DEPTH_TEST);
```
Comment (le-rich) [inline Core/src/main.cpp:124]: Dead code

File: Core/src/main.cpp [hunk 4]
```diff
@@ -157,39 +140,39 @@ int run_glfw()
 
         for (auto system : systems)
         {
-            system->Update(deltaTimeFloatSeconds);
+            //system->Update(deltaTimeFloatSeconds);
```
Comment (le-rich) [inline Core/src/main.cpp:143]: Dead Code

File: Core/src/main.cpp
Comment (le-rich) [inline Core/src/main.cpp:1]: This file has weird indentation

File: Core/include/Components/TransformComponent.h [hunk 5]
```diff
@@ -33,6 +33,7 @@ class TransformComponent : public Component {
 	TransformComponent(Entity* mEntity);
 	// the constructor, takes global position, orientation, scale, parent, child
 	TransformComponent(Entity* mEntity, glm::vec4 position, glm::quat orientation, float scale);
+    TransformComponent(TransformComponent& const);
```
Comment (le-rich) [inline Core/include/Components/TransformComponent.h:36]: Weird indentation

File: Core/include/Scene.h [hunk 6]
```diff
@@ -1,4 +1,6 @@
 #pragma once
+#include <mutex>
```
Comment (le-rich) [inline Core/include/Scene.h:2]: Nitpick: put stdlibs after custom includes

File: Core/src/Components/TransformComponent.cpp
Comment (le-rich) [inline Core/src/Components/TransformComponent.cpp:1]: Inconsistent indentation in this file

File: Rendering/include/Components/CameraComponent.h [hunk 7]
```diff
@@ -6,14 +6,15 @@
...
-	TransformComponent* lookAtTarget;
+	TransformComponent* lookAtTarget = nullptr;
 
     float fieldOfView = 90;
     float farClipPlane = 1000.0f;
     float nearClipPlane = 0.3f;
     float aspectRatio;
 
     CameraComponent(Entity* mEntity);
+    //CameraComponent(CameraComponent const&);
```
Comment (le-rich) [inline Rendering/include/Components/CameraComponent.h:17]: Dead Code

File: Rendering/include/Renderer.h [hunk 8]
```diff
@@ -1,20 +1,23 @@
 #pragma once
+#include <glad/glad.h>
+#include <GLFW/glfw3.h>
```
Comment (le-rich) [inline Rendering/include/Renderer.h:3]: Nitpick, stdlibs after custom includes

**Review by le-rich — CHANGES_REQUESTED** [review]
Very messy accross a lot of files. There's no rush to put this code in so we can take the time to get it right, we can have other people contribute to this branch as well.  Let's not push this in this state.

**Comment by le-rich** [comment]
This PR should be viewed as the multithreading component only, the rendering needs to undergo another task using renderables.

**Review by imnorookie — APPROVED** [review]
:3

**Review by Sepehrman — APPROVED** [review]

**Review by Sepehrman — APPROVED** [review]

**Review by le-rich — APPROVED** [review]

**Pull request merged**